### ISSUE TITLE
BambooMandrillAdapterApiError - Process #PID<> terminating: :timeout

### ISSUE THREAD

**Issue by casio** [thoughtbot/bamboo#299]
Hey guys,

we're experiencing seemingly random errors in our production app with this stacktrace:
``` 
(bamboo) lib/bamboo/adapters/mandrill_adapter.ex:72: Bamboo.MandrillAdapter.deliver/2
(elixir) lib/task/supervised.ex:85: Task.Supervised.do_apply/2
(stdlib) proc_lib.erl:247: :proc_lib.init_p_do_apply/3
``` 

We're on `0.8.0` using the Mandrill adapter. 
Have you maybe seen this elsewhere?
Or maybe, would an attempt to increase the `hackney` timeout options be a feasible attempt to workaround?

Thanks for a pointer!

**Comment by casio** [comment]
fwiw I tried to fix this by configuring `hackney` likes so:
`config :hackney, timeout: 150000, max_connections: 500`

didnt help - still getting the timeouts

however, we're using an umbrella app and I added the config in the app where `hackney` is explicitly started but which is not the app sending the mail. I'll also put it in the other app & report back here

**Comment by paulcsmith** [comment]
@casio any luck? You may want to try the latest beta or use master branch. If I remember correctly we updated hackney. Maybe that'd fix it

**Comment by Preen** [comment]
This is a killer for us :(

**Comment by paulcsmith** [comment]
I haven't seen this so I'm unsure how to fix. If someone finds a fix, please submit a PR for it. @casio Did that fix work for you?

**Comment by kcurtin** [comment]
FYI pretty sure this is an issue in more recent versions of hackney where connections aren't being release.. I've ran into this problem in other elixir packages that depend on hackney.

I'm linking to a PR that attempted to address this but updating hackney did not full resolve the issues in my projects: https://github.com/benoitc/hackney/issues/420

**Comment by egeersoz** [comment]
I run into this frequently when dealing with large attachments. I have a function that sends data to another service and receives a large (15 MB+) PDF file, which I then need to attach to an email and send via Bamboo. I get the PDF file without issues, but Bamboo times out while sending.

**Comment by churcho** [comment]
I am also getting this `(Bamboo.ApiError) :timeout`
Can't find a way around this. I am sending out large attachments.

**Comment by tmarkanov** [comment]
I took the liberty of forking the repo to resolve this issue, as I have had the same problem.
https://github.com/tmarkanov/bamboo

**Comment by maymillerricci** [comment]
Just merged #498 which allows you to pass in `hackney_opts` for example for specifying timeouts. Hopefully this should resolve this! Feel free to re-open if it does not.
